src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -12,8 +12,8 @@ public static class Enumerable
     {
         public static IEnumerable<TSource> Where<TSource>(this IEnumerable<TSource> source, Func<TSource, bool> predicate)
         {
-            if (source == null) throw Error.ArgumentNull("source");
-            if (predicate == null) throw Error.ArgumentNull("predicate");
+            Error.CheckSourceNotNull(source);
+            Error.CheckPredicateNotNull(predicate);
```

Comment:
I'd actually thought it would make things worse, as I've seen the JIT refuse to inline methods that contain ```throw```, and I thought it would refuse to inline these helpers because of that, thus we'd actually be increasing the number of function calls that couldn't be inlined.  But, I just checked and tried it out, and it appears at least the VS2015 x86 and x64 JITs are able to inline methods with throws in some cases, just not in others (e.g. https://github.com/dotnet/coreclr/blob/master/src/jit/importer.cpp#L12588-L12612), and these helpers are actually able to be inlined, at least in my experiments.  

In any event, thanks for separating them out.  Since we'd be changing such error checking paths, I'd also hope to see tests in place that would exercise all of those checks being changed.